Fix part of #21448: Colorize terminal messages for acceptance tests

souja450 opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes  #21448.
2. This PR does the following: [This PR introduces colorization for log messages in CI output, improving readability during checks. Specifically:
Error tracebacks are displayed in red.
Success messages are displayed in green.
Other message types (info, warning, debug) are also supported for better clarity.
This change improves developer experience and addresses feedback from [#21448](https://github.com/oppia/oppia/issues/21448)
.]
3. (For bug-fixing PRs only) The original bug occurred because: [The issue occurred because error and success messages were not colorized in the CI logs, making them harder to distinguish.
This PR resolves that by introducing a centralized log_to_terminal() utility that applies color formatting using ANSI escape codes.]

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [X] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [X] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [X] I have written tests for my code.
- [X ] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [X] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Proof that changes are correct

<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network
- The Error messages are colorized in **red**:

https://github.com/user-attachments/assets/5b0e68fa-5164-4ad7-bbb9-24526becb693


-  The Success messages are colorized in **green**:

https://github.com/user-attachments/assets/37f9bb1b-a1e3-4a53-8a94-c13be4ea0e7c
<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling


- The Error messages are colorized in **red**:

https://github.com/user-attachments/assets/5f0441b4-b8b1-4cfd-b06b-3f0789069c0f

-  The Success messages are colorized in **green**:

https://github.com/user-attachments/assets/37f9bb1b-a1e3-4a53-8a94-c13be4ea0e7c


-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Add terminal logging and acceptance test helpers to scripts/common.py
- Introduce LogType enum and log_to_terminal() for colored logging output.
- Update acceptance test utilities: list_acceptance_tests, get_acceptance_test_path, run_acceptance_test.
- Fix imports, type hints, and ensure Python 3 compatibility.
- Refactor code to pass pylint checks and maintain 80-character line limits.

souja450 committed:
Add terminal logging and acceptance test helpers to scripts/common.py
- Introduce LogType enum and log_to_terminal() for colored logging output.
- Update acceptance test utilities: list_acceptance_tests, get_acceptance_test_path, run_acceptance_test.
- Fix imports, type hints, and ensure Python 3 compatibility.
- Refactor code to pass pylint checks and maintain 80-character line limits.

souja450 committed:
Add terminal logging and acceptance test helpers to scripts/common.py
- Introduce LogType enum and log_to_terminal() for colored logging output.
- Update acceptance test utilities: list_acceptance_tests, get_acceptance_test_path, run_acceptance_test.
- Fix imports, type hints, and ensure Python 3 compatibility.
- Refactor code to pass pylint checks and maintain 80-character line limits.

souja450 committed:
Update dev requirements

souja450 committed:
Add terminal logging and acceptance test helpers to scripts/common.py
- Introduce LogType enum and log_to_terminal() for colored logging output.
- Update acceptance test utilities: list_acceptance_tests, get_acceptance_test_path, run_acceptance_test.
- Fix imports, type hints, and ensure Python 3 compatibility.
- Refactor code to pass pylint checks and maintain 80-character line limits.

souja450 committed:
Add colorized logging for error to common.py

souja450 committed:
Edit common.py for colorized logging and update requirements_dev.txt via install script

souja450 committed:
Fix: Replace print statements with sys.stdout.write in common.py

Oppia’s lint checks do not allow the use of print statements.
Replaced print with sys.stdout.write in scripts/common.py at
lines 295–330, 516–544, 660–661, 787–858, 961, and 1021 to
comply with the linting rules.

souja450 committed:
Add unit tests for log_to_terminal in common_test.py

souja450 committed:
Add unit tests for log_to_terminal in common_test.py

souja450 committed:
Add unit tests for log_to_terminal in common_test.py

souja450 committed:
Add unit tests for log_to_terminal in common_test.py

souja450 committed:
Add unit tests for log_to_terminal in common_test.py

souja450 committed:
Ignore datastore-emulator-data directory in git

souja450 committed:
To fix log_to_terminal tests and update coverage configuration

souja450 committed:
Add log_to_terminal helper and unit tests

- Introduced log_to_terminal function with color-coded output.
- Added corresponding tests in common_test.py to ensure correct behavior.
- Updated scripts/common.py to integrate the helper.

Note: Linter may flag existing print statements in common.py, but these were not introduced in this PR.

souja450 committed:
Add unit test to cover debug branch in log_to_terminal

souja450 committed:
Add tests for OS helpers and ensure_directory_exists in common.py

souja450 committed:
Fix RequireCwdToBeOppiaTests to use assertRaisesRegex instead of assertRaises

souja450 committed:
Fix RequireCwdToBeOppiaTests to use assertRaisesRegex instead of assertRaises

souja450 committed:
Fix RequireCwdToBeOppiaTests to use assertRaisesRegex instead of assertRaises

souja450 committed:
Fix RequireCwdToBeOppiaTests to use assertRaisesRegex instead of assertRaises

souja450 committed:
Fix RequireCwdToBeOppiaTests to use assertRaisesRegex instead of assertRaises

souja450 committed:
Fix lint issues in scripts/common_test.py to support error colorization changes

souja450 committed:
Fix lint issues in scripts/common_test.py to support error colorization changes

mon4our committed:
Fix #23249: Added error message for when skill is not selected while adding a skillreview component (#23252)

Added error message for when skill is not selected

KartikSuryavanshi committed:
Fix part of #10700:Migrated StatesObjectFactory to States (#23185)

* Fix part of #10700:Migrated StateObjectFactory to State

* Made changes

* Made changes

* Fix part of #10700:Migrated StatesObjectFactory to States

* Made changes

* Fixed type checks

* made changes

seanlip committed:
Add maximum time limits for running long tests. (#23322)

KartikSuryavanshi committed:
Fix part of #10700:Migrated QuestionObjectFactory to Question (#23186)

* Fix part of #10700:Migrated StateObjectFactory to State

* Made changes

* Made changes

* Fix part of #10700:Migrated QuestionObjectFactory to Question

* Renamed file

* Renamed file

* Fixed lint checks

* Made changes

* Fix #22817: Fix Elasticsearch under docker (#23078)

* Fix Elasticsearch under docker

Elasticsearch stopped working under docker due to PR #20809 because of a
bug in how we checked whether we were running in a local development
environment when creating an Elasticsearch client.

* Upgrade elasticsearch

* Add common.py to cache key to trigger elasticsearch re-install

* Fix part of #10700:Migrated WrittenTranslationsObjectFactory to WrittenTranslations (#23161)

* Migrated WrittenTranslationObjectFactory to  WrittenTranslation

* made changes

* Fixed failing checks

* Made changes

* Made changes

* Made changes

* fixed lint checks

* Made changes requested

* Fix part of #10700:Migrate WrittenTranslationsObjectFactory to WrittenTranslations

* Delete core/templates/domain/exploration/lost-change.model.ts

Got committed by mistake

* Resolved merge conflicts

* Fixed frontend failing tests

* Update some Python libraries (#23180)

* Update some libraries

* Fix issues arising due to upgrade of datastore library.

* Fix typo

* Delete unused functions (that were also causing errors with new datastore update

* Fix issues with not-equal query

* Fix mypy and backend test failures.

* Revert "Update some Python libraries (#23180)"

This reverts commit b47390e11c62349e3877084a60589bb098b806c7.

* Revert "Fix part of #10700:Migrated WrittenTranslationsObjectFactory to WrittenTranslations (#23161)"

This reverts commit acffef1b4fbf8735c06d7ecdcd1406c24e9f26de.

* Revert "Fix #22817: Fix Elasticsearch under docker (#23078)"

This reverts commit 8ed1986e5b277d516a51f2591400c73a63a9c22c.

---------

Co-authored-by: U8N WXD <U8NWXD@users.noreply.github.com>
Co-authored-by: Sean Lip <sean@seanlip.org>

NITISH084 committed:
FIX -#22711, #23140 : Translator is not able to go to next card after submitting translation (#23207)

* Testing

* made few changes

* made changes

* made changes

* made changes in spec

* made changes in spec

* made changes in spec

---------

Co-authored-by: NITISH084 <imnitishkmar04@gmail.com>

kevintab95 committed:
Add dummy beam jobs for testing purposes (#23294)

* add dummy beam jobs

* add test files to backend_test_shards.json

* fix lint errors

---------

Co-authored-by: Kevin Thomas <kevintab@mbp.local>

KartikSuryavanshi committed:
Fix part of #10700:Migrated ExplorationObjectFactory to Exploration (#23190)

* Fix part of #10700:Migrated StateObjectFactory to State

* Made changes

* Made changes

* Fix part of #10700:Migrated StatesObjectFactory to States

* Made changes

* Fix part of #10700:Migrated ExplorationObjectFactory to Exploration

* Made changes

* Fixed lint checks

* Fixed type checks

* Made changes --no-verify

* Made changes

* Delete requirements_dev.in

* Delete requirements_dev.txt

* Removed mistakenly added files

* Made changes

mon4our committed:
[GSoC 2025] M2 - Fix part of #19851: Update rte_components to rte_component_config_id (#23329)

Updated property name

NITISH084 committed:
Fix #23015 : Ensure Practice Tab visibility matches learner view in topic preview (#23170)

* Linked the checkbox and practice tab

* topic-viewer-page updated

* updated layout of practice-component in both topic-preview tab and topic-viewer-page

* resolved Lint errors

* Added the translation pipeline

* resolved Frontend-test errors

* Added test in topic-preview-tab.component.spec.ts

* resolved coverage issues

* Solved linter errors

* resolved coverage issues

* resolved coverage issues

* resolved coverage issues

* resolved coverage issues

* resolved coverage issues

* Updated the topic-editor-state -service.ts

* updated spec file

* updated coverage

* updated coverage

* reverted useless changes

* reverted topic-editor-state service spec

* removed unnecessary changes

---------

Co-authored-by: NITISH084 <imnitishkmar04@gmail.com>

chris7716 committed:
Fixes part of #22793:Add a method to find the content of a given content ID (#23283)

* service file name changed to backend-api

* backend api file added

* previous service file removed from CODEOWNERS

* new backend service file added to  CODEOWNERS

* Python file change

* Rollback

* First pass

* Resolve tests

* Remove unwated test

* Fix lint

NITISH084 committed:
Fixes #20924: Add warning modal on exiting Question editor with unsaved changes (#23184)

* Doing tests

* updated for topic-editor

* Revert "updated for topic-editor"

This reverts commit 2387f4fb6f207d0e8acdc9569d9b1888ce97a0a8.

* Updated for mobile views on topic editor

* Done for desktop mode

* Done for mobile view skill editor

* removing unnecessary changes

* Resolved Lint errors

* Updated funcn name and added core/templates/pages/skill-editor-page/navbar/skill-editor-navbar.component.spec.ts test for this file

* resolved few lint issues and added coverage tests

* resolved lint errors

* Added warning even for closing and reloading tabs

* updated topic-navbar spec

* Resolved prev mistake

* updated tests

* Trigger CI

* updated createQuestion funcn

---------

Co-authored-by: NITISH084 <imnitishkmar04@gmail.com>

seanlip committed:
Upgrade as many 3rd party python libraries as possible, and drop some that are no longer used. (#23279)

* Update remaining Python libraries and drop unneeded ones.

* fix mismatch logic

* fix dev deps

* Cleanup

* exclude direnv from mypy checks

* fix mismatch

* fix mismatch

* fix normalization

* lint

* Fix tests

* Drop requirement for certifi in PROD. Drop explicit google-cloud-bigquery pinning.

* Fix more tests

* Add pytest since our pylint_extensions_test requires it

* Fix gcloudignore

* Do not ignore webpack_bundles

* Update proto-plus

* Make tests more robust

* Upgrade to proto-plus==1.25.0

* Fix filepath for package.json in platform_parameter_services.py

* Fix test

* Fix lint

HardikGoyal2003 committed:
Fix #7222: Remove Upgraded services file (#23280)

* Removed upgraded services

* Removed import Todos

* Removed instances of services

NITISH084 committed:
Fixes #23332 : Corrects the anchor link for the "BROWSE OUR LESSONS" button on Home tab in learner dashboard (#23360)

added communityLibraryUrl

Co-authored-by: NITISH084 <imnitishkmar04@gmail.com>

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

HardikGoyal2003 committed:
Fix #13080: Remove angularjs mock (#23281)

* Removed mock-ajs.ts

* Removed instances of mock-ajs file

* MyPy Update

* Removed angularjs-simple-logger instance

* Revert "MyPy Update"

This reverts commit ae1f884cf62589e99d2ae06e879f4075876fe1cf.

NITISH084 committed:
Fixes - 23264, 23268, 23272 : Resolve Learner Dashboard responsive issues and console errors (#23288)

* ICU Message format corrected

* Made multiple changes

* Lint errors

* added tests

* Lint errors

* Lint errors

* Updated spec file according to pr changes

* updated acc to instruction

* Added new translation key

---------

Co-authored-by: NITISH084 <imnitishkmar04@gmail.com>
Co-authored-by: Mohit Ruwatia <120438312+mon4our@users.noreply.github.com>

Helper2020 committed:
Black formatter pre-stage (#23089)

* set isort profile to black

* Modifed pylintrc

* Added black to requirements

* Add pyproject

* Added pyproject.toml

* Added newline to pyproject.toml

* ran isort fixed linting errors

* Fixed linting errors

* ran isort on missed files

* Removed  vscode search file

* Moved dependeny to correct alphetbtical position

* Removed some pylint disables

* Removed uncessary pylint disable comment

* Removed trailing whitespace.

* Removed duplicate line

* Added linter to detect disabling Black formatter

* Fixed linting errors

* Added testing files to files to files without associated test files

* Added multi-strings explaining newly added testing files

* Remove generate build directory files'

* Removed shard

* Removed corresponding parts in make file

* Put back original code in start.py

* ran isort on scripts/start.py

* ran isort on cloud_datestore_services.py

* Undueing changes in installing_third_party_libs

* Reversed isort changes in setup_test.py

* Removed uncessary pylint directive

* update requirements

* Fixed linting error after update

jayam04 committed:
[GSoC 2025 - Acceptance Tests] M2.8 (Fixes part of #22716) - Implement Acceptance tests for RC, SM, and SA (from CUJ v3 Doc). (#23262)

* Added acceptance tests for release coordinator

* Revert It: Removed other test files

* Added acceptance tests for superAdm

* Added acceptance tests for site moderator

* Fixed navigate to release co..

* Few changes

* Fix: Type Errors

* Fix: Minor issues in each of the test

* Fix: Fixes platform param not working in some cases

* Fix: Fixes more test steps

* Fix: Fixes some tests

* Fix: Fixes error message in one test, comparasion check in another

* Fixes minor test failures

* Minor Fixes

* Fix: Updates proper timestamp comparasion

* Fix: Updates proper timestamp comparasion

* Fixes 2 tests

* Fixes the last test in mobile

* Updates full_stack_tests.yml

* Update full_stack_tests.yml

* Addressed most of the comments.

* Adds screenshots

* Addresses the remaining comments

* Revert "Revert It: Removed other test files"

This reverts commit c2e257d23a9ef01352c988523425e97910ac9cf9.

* Reverts full_stack_tests.yml

* Removes changes not of mine from full_stack_tests.yml

* Removes changes not of mine from full_stack_tests.yml

* Addresses Comments

* Fixes some type errors

* Fix: beamJobStatusSelectorPrefix (two '.' before className)

* Potential Fix: Old flake navigation timeout.

* Fix: TypeErrors

mon4our committed:
Fix #23250: Fixed Rich Text Editor display error on Contributor Dashboard (#23367)

* Fixed rte error

* Made requested changes

* Updated screenshot

* Updated screenshots

seanlip committed:
Fix issue with pip tools install changing the command in requirements_dev.txt (#23410)

* Print requirements diff info

* Fix tests

* unfix tests

* unfix tests

* Drop unnecessary check if file is not changed materially.

* Print contents of requirements_dev.txt and requirements.txt in downloaded artifact

* Add more debug info

* Fix template

* Fix template

* Fix logic

piyush140104 committed:
Fix #23346 : Artifact name for mobile test failures (#23368)

Update artifact name for mobile test failures

Nik-09 committed:
Handles unknown rte tags during voiceover regeneration (#23232)

* Upgrades elastic search version

* Add checks to handle unknown tags

* Adds backend tests

* Fixes backend tests

* Addresses review comments

* Adds custom tags mapping

* Fixes mypy checks

* Adds HTML validation

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
Tests: Add unit tests for error/success colorized logging in common.py

souja450 committed:
fix: update color handling in common.py and add related tests in common_test.py

souja450 committed:
fix: update color handling in common.py

souja450 committed:
fix: update color handling in common.py

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 commented:
Hi @seanlip, they are seen now on the top of the webpage. Thank you.

souja450 commented:
> > Hi @seanlip, they are seen now on the top of the webpage. Thank you.
> 
> Hi @seanlip, they are seen now on the top of the webpage. Thank you.

Hi @seanlip, they are seen now on the top of the webpage. Thank you.

jayam04 commented on scripts/common.py:
```diff
@@ -986,3 +987,53 @@ def start_subprocess_for_result(cmd: List[str]) -> Tuple[bytes, bytes]:
+    LogType.ERROR: '\033[91m',
+    None: ''
+}
+_END_COLOR = '\033[0m'
```
@souja450 wouldn't it be better if we define it as class variable in class LogType. Then, we can use LogType.COLOR.INFO

jayam04 commented on scripts/common_test.py:
```diff
@@ -1474,3 +1476,156 @@ def mock_urlopen(
                 self.assertEqual(buffer.read(), b'content')
+
+
+class CommonSwapEnvironTests(test_utils.GenericTestBase):
```
Is this test related to your changes?

jayam04 commented on scripts/common_test.py:
```diff
@@ -1474,3 +1476,156 @@ def mock_urlopen(
+        self.assertNotIn('BAZ', os.environ)
+
+
+class CommonBranchTests(test_utils.GenericTestBase):
```
Same as above, is this related?

jayam04 commented on scripts/common_test.py:
```diff
@@ -1474,3 +1476,156 @@ def mock_urlopen(
+            self.assertFalse(common.is_current_branch_a_release_branch())
+
+
+class CommonGitTests(test_utils.GenericTestBase):
```
Same, is this related.

jayam04 commented on scripts/common_test.py:
```diff
@@ -1474,3 +1476,156 @@ def mock_urlopen(
+                common.get_remote_alias(['invalid'])
+
+
+class CommonLogToTerminalTests(test_utils.GenericTestBase):
```
Are all tests under this class covered by tests under `LogToTerminalTests`

jayam04 reviewed (changes requested):
@souja450 Thanks for the PR. I would definately love to see my logs more colorful.

I have left some comments. Also, sorry for the delay, I was out of office.

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Fix part of #21448: Colorize terminal messages and move color mapping to LogType

souja450 committed:
Merge branch 'fix-color-color' of github.com:souja450/oppia into fix-color-color

souja450 commented:
Hi @jayam04,
Thank you for the detailed review. I’ve addressed all your comments as follows:

- I moved the color codes into a nested COLOR class within LogType, allowing them to be accessed as LogType.COLOR.INFO, LogType.COLOR.ERROR, etc.

- I removed unrelated test classes (CommonSwapEnvironTests, CommonBranchTests, and CommonGitTests) that were unintentionally included during a previous merge, to keep this PR focused on log colorization.

- I verified that all relevant tests for the logging functionality are properly covered under LogToTerminalTests.

Apologies for the delay in addressing your review, and thank you for your patience.
All lint, type, and backend tests are now passing successfully.

souja450 commented:
@Reviewer, PTAL. Thanks!

souja450 committed:
Fix part of #21448: Add colored error message

souja450 committed:
Fix part of #21448: Add colored error message

souja450 committed:
Fix part of #21448: Add colored error message

souja450 committed:
Fix part of #21448: Add colored error message

souja450 committed:
Fix part of #21448: Add colored error message

souja450 commented:
Hi @jayam04,
I made a few improvements to the code after your last feedback. Everything was passing before I pushed these updates, but now I’m seeing a couple of CI failures (I believe they’re likely due to recent changes in develop). My local checks are passing fine.
I’ll wait for your review before doing any further updates. Thanks!

jayam04 commented on scripts/common.py:
```diff
@@ -986,3 +989,157 @@ def start_subprocess_for_result(cmd: List[str]) -> Tuple[bytes, bytes]:
+    ERROR = '\033[91m'
+
+    # Reset color.
+    END = '\033[0m'
```
@souja450 Why create a separate class and attach it afterwards. Would it be better if we use static dict for color inside LogType class.

So,
```
class LogType:
    COLOR = {...}
```

This way, we can also remove no MYPY used many times in below code.

jayam04 commented on scripts/run_acceptance_tests.py:
```diff
@@ -184,4 +184,9 @@ def main(args: Optional[List[str]] = None) -> None:
+    except Exception:
+        # Use Oppia's colored traceback instead of the default.
+        common.print_colored_traceback()
+        sys.exit(1)
```
Won't it print error logs twice? If without color and another with color?

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

souja450 committed:
Fix: common.py

souja450 committed:
Fix: common.py

souja450 committed:
Fix: common.py

souja450 committed:
Merge branch 'develop' into fix-color-color

jayam04 commented:
Hi @souja450! First of all, thanks for your contributions. I want to let you know that if you’re facing any challenges or need assistance with something, feel free to reach out to me at [oppia@jayampatel.space](mailto:oppia@jayampatel.space)!

jayam04 commented on scripts/common.py:
```diff
@@ -1029,3 +1032,134 @@ def start_subprocess_for_result(cmd: List[str]) -> Tuple[bytes, bytes]:
+
+def print_colored_traceback() -> None:
+    """Prints the most recent exception traceback in red color."""
+    import traceback  # pylint: disable=import-outside-toplevel
```
@souja450 Is there any specific reason that we are importing traceback new and not top of the file.

souja450 replied:
> @souja450 Is there any specific reason that we are importing traceback new and not top of the file.

Hi @jayam04,
I kept the traceback import inside the function because importing it at the top of the file caused a lint error. The file already has an __exit__ method that uses a traceback parameter.
Keeping the import inside the function prevents that naming conflict, and now all lint and test checks are passing successfully.

jayam04 replied:
@U8NWXD Can you help what we should do in this case?

souja450 replied:
Hi @jayam04, I moved the traceback import back to the top of the file and adjusted the code so that lint passes successfully now. Thanks for pointing it out!

U8NWXD replied:
My preference would be to change the `exit()` method parameter to something other than `traceback` so that we can keep the canonical `traceback` import

jayam04 reviewed (changes requested):
@souja450 Just one comment. And also, update the PR description and remove [] brackets from Issue number - `This PR fixes or fixes part of #[21448].` to `This PR fixes #21448.`

souja450 committed:
Common.py edited

souja450 committed:
Common.py edited

jayam04 commented:
@U8NWXD Can you help with the following [comment](https://github.com/oppia/oppia/pull/23276#discussion_r2540482985).

souja450 committed:
Common.py edited

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

jayam04 commented:
@souja450 Can you address the previous comment based on @U8NWXD's suggestion. https://github.com/oppia/oppia/pull/23276#discussion_r2548110624

souja450 commented:
> @souja450 Can you address the previous comment based on @U8NWXD's suggestion. [#23276 (comment)](https://github.com/oppia/oppia/pull/23276#discussion_r2548110624)

Hello @jayam04  thank you for the feedback. Yes, I can fix it according to @U8NWXD comment.

souja450 committed:
Common.py edited

souja450 commented:
> @souja450 Can you address the previous comment based on @U8NWXD's suggestion. [#23276 (comment)](https://github.com/oppia/oppia/pull/23276#discussion_r2548110624)

I fixed it now according to @U8NWXD and now I pushed the changes.

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

jayam04 reviewed (approved):
LGTM

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
merge conflict issue

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color

souja450 committed:
Merge branch 'develop' into fix-color-color
